samtranslator/translator/translator.py

```diff
@@ -269,6 +274,83 @@ def _get_resources_to_iterate(
         return functions + statemachines + apis + others + connectors
 
 
+def add_embedded_connectors(sam_template: Dict[str, Any], document_errors: List[Any]) -> None:
+    """
+    Loops through the SAM Template resources to find any connectors that have been attached to the resources.
+    Adds those attached connectors as Connector resources to the list of resources
+
+    :param dict sam_template: SAM template
+    """
+    resources = sam_template.get("Resources", {})
+    connectors = {}
+
+    # Loop through the resources in the template and see if any connectors have been attached
+    for source_logical_id, resource in resources.items():
+        if "Connectors" in resource:
+            try:
+                sam_expect(
+                    resource["Connectors"], source_logical_id, "Connectors", is_resource_attribute=True
+                ).to_be_a_map()
+
+                # Go through each of the connectors that have been attached and create a Serverless Connector resource
+                for connector_logical_id, connector_dict in resource["Connectors"].items():
+
+                    # The logical id for the transformed connector will be <source_logical_id> + <embedded_connector_logical_id>
+                    connector_logical_id = source_logical_id + connector_logical_id
+
+                    # can't use sam_expect since this is neither a property nor a resource attribute
```

Comment:
nit: Maybe it's just me, but it's great to have comments. However, I feel like there are too many comments that's kinda make the code readability a bit hard. Plus, these codes are pretty straightforward to understand without comments.